docs/release.md

```diff
@@ -0,0 +1,77 @@
+# OSA Release process
+
+OSA Release process consist of multiple dimensions:
+* Plugin code release
+* Config release
+* VM image release
+
+Update might be one or any combination of all dimensions. But individual upgrade path is validated using adminAPI. 
+Plugin code release is a new branch and considered a major release. Any VM image or config change is a minor release.
+
+# Versioning
+
+## Major releases
+
+Major release creation flow:
+1. Create a branch with a plugin code you wish to release.
+2. Commit to update `PluginConfig` referencing release image versions.
+3. Git tag major release version (`v3.0`)
+4. Build release images (sync, metricsbridge, etc)
+5. Make sure VM image is released and is available to be used by plugin code.
+6. Update docs & release notes
+
+## Minor releases
+
+If there is a need to update already released code - minor release is done.
+This might happen if we make a mistake, want to update configuration, or not code related fix.
```

Comment:
Can you give me an example of a "not code related fix"?

Reply:
I think all non code related fix can be put under 'configuration' section